modules/binding-infra-node-workloads-using-taints-tolerations.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+
+[id="binding-infra-node-workloads-using-taints-tolerations_{context}"]
+= Binding infrastructure node workloads using taints and tolerations
+
+If you have an infra node that has the `infra` and `worker` roles assigned, you
+must configure the node so that user workloads are not assigned to it.
+
+.Prerequisites
+
+* Configure additional MachineSets in your {product-title} cluster.
+
+.Procedure
+
+. Use the following command to add a taint to the infra node to prevent
+scheduling user workloads on it:
++
+[source,terminal]
+----
+$ oc adm taint nodes <node-name> <key>:<effect>
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc adm taint nodes node1 node-role.kubernetes.io/infra:NoSchedule
+----
++
+This example places a taint on `node1` that has key
+`node-role.kubernetes.io/infra` and taint effect `NoSchedule`. Nodes with the
+`NoSchedule` effect schedule only Pods that tolerate the taint, but allow existing
+Pods to remain scheduled on the node.
++
+[NOTE]
+====
+If a descheduler is used, Pods violating node taints could be evicted from the
+cluster.
+====
+
+. Add tolerations for the Pod configurations you want to schedule on the infra
+node, like router, registry, and monitoring workloads. Add the following code to
+the Pod specification:
++
+[source, yaml]
+----
+tolerations: 
+  - effect: NoSchedule <1>
+    key: node-role.kubernetes.io/infra <2>
+    operator: Exists <3>
```

Comment:
Can you set this directly on router pods?  I'd expect the Deployment controller to stomp your changes.  And then if you touch the router Deployment, I'd expect the ingress operator to stomp your changes.  I'd expect you'd have to touch [IngressController][1] to explain to the ingress operator the tolerations and nodePlacement you wanted, and the ingress operator would set the Deployment and the Deployment controller would set the Pod.

It may also be similar for the registry and monitoring, I'm not sure.  And maybe I'm wrong, and one of the controllers doesn't mind folks leaning in and poking at this sort of thing in its operands.

[1]: https://docs.openshift.com/container-platform/4.6/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-router_creating-infrastructure-machinesets

Reply:
The infra components list (including the default router) was added/approved back for OCP 4.1. So I assume this worked at *some point*. But I definitely think we need to either retest these components or get someone to provide us with an already-test approved list of infra components.

At this point, we've solidified a way to ensure workload scheduling to infra nodes, and confirmed infra nodes with the `worker` label are not counted towards a subscription, which were two of the biggest issues expressed in the BZ. I see approving/verifying the infra component list as the next step. With the pace at which we're currently moving, I'm leaning towards merging this content as-is and introducing additional changes for the infra component list once we can track down more info.